client-tools/Shell/RequestFuzzer.h

```diff
@@ -0,0 +1,190 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2022 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Julia Puget
+////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include <array>
+#include <charconv>
+#include <deque>
+#include <map>
+#include <optional>
+#include <random>
+#include <string>
+#include <string_view>
+#include <unordered_map>
+#include <unordered_set>
+
+#include "Basics/VelocyPackHelper.h"
+
+namespace arangodb {
+namespace fuzzer {
+
+enum CharOperation { ADD_STRING, ADD_INT_32, MAX_CHAR_OP_VALUE };
+
+enum LineOperation {
+  COPY_LINE,
+  INJECT_RAND_BYTE_IN_LINE,
+  ADD_LINE,
+  MAX_LINE_OP_VALUE
+};
+
+enum RequestType {
+  GET,
+  PUT,
+  PUT_RAW,
+  OPTIONS,
+  POST,
+  PATCH,
+  DELETE,
+  HEAD,
+  MAX_REQ_VALUE
+};
+
+enum BodyOperation { ADD_ARRAY, ADD_OBJECT, ADD_CHAR_SEQ, MAX_BODY_OP_VALUE };
+
+enum class HttpWord { HTTP, RAND_HTTP_WORD, MAX_VALUE };
+
+enum class HttpVersion { ACTUAL, RAND_HTTP_VERSION, MAX_VALUE };
+
+class RequestFuzzer {
+  struct RandContext {
+    RandContext(uint32_t seed) : mt(seed) {}
+
+    std::mt19937 mt;
+    static constexpr uint32_t maxRandAsciiStringLength = 50;
+  };
+
+  struct Result {
+    std::string restVerb;
+    std::string location;
+  };
+
+ public:
+  RequestFuzzer(std::optional<uint32_t> numIt, std::optional<uint32_t> seed)
+      : _numIterations(numIt.value_or(limitNumIterations)),
+        _seed(seed.value_or(std::random_device()())),
+        _randContext{_seed},
+        _stringLines{},
+        _keysAndValues{},
+        _reqHasBody(false) {}
+
+  void randomizeHeader(std::string& header, std::optional<size_t> payloadSize);
+  std::optional<std::string> randomizeBody();
+  void randomizeBodyInternal(velocypack::Builder& builder);
+  // void randomizeReq(std::string& body, std::string& header, bool& hasBody);
+  void logStart();
+
+ private:
+  void randomizeCharOperation(std::string& input, uint32_t numIts = 1);
+
+  void randomizeLineOperation(uint32_t numIts = 1);
```

Comment:
I think the default parameter values should be removed. The caller should explicitly specify the number of iterations.